### PR TITLE
boards: add hpmicro hpm6750evkmini board

### PR DESCRIPTION
**Pull request by Cherish-Gww**
This is an risc-v board. we also add hpmicro SoC support, as well as gpio, pinctrl, uart driver.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
this PR needs to be split into meaningful commits.

**Comment by Cherish-Gww** [comment]
> this PR needs to be split into meaningful commits.

> this PR needs to be split into meaningful commits.

split done, I have tested zephyr/samples/hello_world/ on my platform, it runs ok.

File: dts/bindings/gpio/hpmicro,hpm-gpio.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      GPIOA (0UL)
+      GPIOB (1UL)
+      GPIOC (2UL)
+      GPIOD (3UL)
+      GPIOE (4UL)
+      GPIOF (5UL)
+      GPIOX (13UL)
+      GPIOY (14UL)
+      GPIOZ (15UL)
+    required: true
```
Comment (fkokosinski) [inline dts/bindings/gpio/hpmicro,hpm-gpio.yaml:27]: This property doesn't seem to be used anywhere in the GPIO driver?

File: dts/riscv/hpmicro/hpm6750.dtsi [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cpu0: cpu@0 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <0>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
```
Comment (fkokosinski) [inline dts/riscv/hpmicro/hpm6750.dtsi:23]:
```suggestion
			riscv,isa = "rv32gc_xandes";
```

File: dts/riscv/hpmicro/hpm6750.dtsi [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+		cpu1: cpu@1 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <1>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
```
Comment (fkokosinski) [inline dts/riscv/hpmicro/hpm6750.dtsi:37]:
```suggestion
			riscv,isa = "rv32gc_xandes";
```

File: include/zephyr/drivers/clock_control/hpmicro_clock_control.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/dt-bindings/clock/hpmicro-clocks.h>
+
+/**
+ * @brief Generate clock setting preset
+ *
+ */
+#define HPM_MAKE_SYSCTL_PRESENT(x)		BIT(x)
```
Comment (fkokosinski) [inline include/zephyr/drivers/clock_control/hpmicro_clock_control.h:21]: Looks like a redundant macro redefinition

File: west.yml [hunk 5]
```diff
@@ -170,6 +170,11 @@ manifest:
       path: modules/hal/gigadevice
       groups:
         - hal
+    - name: hal_hpmicro
+      revision: pull/2/head
+      path: modules/hal/hpmicro
+      groups:
+        - hal
```
Comment (fkokosinski) [inline west.yml:177]: Looks like you're missing an entry in the `MAINTAINERS.yml` file for this HAL.

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: dts/bindings/gpio/hpmicro,hpm-gpio.yaml (hunk 1 above)
Comment (Cherish-Gww) [reply dts/bindings/gpio/hpmicro,hpm-gpio.yaml:27]: > This property doesn't seem to be used anywhere in the GPIO driver?

In gpio_hpmicro.c L237:
`		.port_base = DT_INST_PROP(n, hpmicro_gpio_port),	\`

**Comment by Cherish-Gww** [comment]
@fkokosinski @gmarull 
If you have time, please help me review this patch. Thank you very much.

**Review by fkokosinski — DISMISSED** [review]

**Comment by Cherish-Gww** [comment]
@fkokosinski 
sorry, I resolved a conflict that dismissed your approve

**Comment by Cherish-Gww** [comment]
@gmarull 
Do you have time to review the patch?

**Review by fkokosinski — DISMISSED** [review]

**Comment by Cherish-Gww** [comment]
@MaureenHelm 
Could you help me review this PR?

File: dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  &pinctrl {
+    pinmux_uart0: pinmux_uart0 {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PORTY, 7,IOC_TYPE_IOC, 0, 2)>,
+        <HPMICRO_PINMUX(HPMICRO_PORTY, 6,IOC_TYPE_IOC, 0, 2)>;
```
Comment (gmarull) [inline dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: suggestion here: to simplify macro usage, make this `HPMICRO_PINMUX(Y, 7, IOC, 0, 2)>`, and use e.g. `HPMICRO_PORT ## port` in the `HPMICRO_PINMUX` macro

Comment (Cherish-Gww) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: Done

Comment (gmarull) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: update docs?

Comment (Cherish-Gww) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: > update docs?

Is update the use of Pinmux usage to the doc？

Comment (Cherish-Gww) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: > update docs?

https://github.com/zephyrproject-rtos/zephyr/compare/bc99ea109892652d625fc73c316b105c12a9ef15..03c9348b8abc6a9f5f8680e4a5868bbbdea5b2f4

**Review by gmarull — CHANGES_REQUESTED** [review]
you're adding a board before it can be built, please, make sure stuff is bisectable

**Comment by Cherish-Gww** [comment]
> you're adding a board before it can be built, please, make sure stuff is bisectable

I don't quite understand this sentence, what should I do for this PR?

**Comment by fkokosinski** [comment]
> > you're adding a board before it can be built, please, make sure stuff is bisectable
> 
> I don't quite understand this sentence, what should I do for this PR?

Hi! The problem is that if you check out on this commit 569fed81d171ed22d300da4d0941fe1a57878f3e you can't build the board it adds. This potentially may give you a false-negative when bisecting.

You should reorder your commits so that their dependencies are added earlier, e.g. SoC is added before the board.

**Comment by Cherish-Gww** [comment]
> > > you're adding a board before it can be built, please, make sure stuff is bisectable
> > 
> > 
> > I don't quite understand this sentence, what should I do for this PR?
> 
> Hi! The problem is that if you check out on this commit [569fed8](https://github.com/zephyrproject-rtos/zephyr/commit/569fed81d171ed22d300da4d0941fe1a57878f3e) you can't build the board it adds. This potentially may give you a false-negative when bisecting.
> 
> You should reorder your commits so that their dependencies are added earlier, e.g. SoC is added before the board.

is that ok?

**Review by fkokosinski — DISMISSED** [review]

**Comment by Cherish-Gww** [comment]
@gmarull 
https://github.com/zephyrproject-rtos/zephyr/pull/64117/commits
is that ok?

**Comment by Cherish-Gww** [comment]
@gmarull Do you have time to help me take a look?

File: dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml (hunk 6 above)
Comment (gmarull) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml:24]: this still looks outdated vs what's on board-pinctrl.dtsi

Comment (Cherish-Gww) [reply dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml]: > this still looks outdated vs what's on board-pinctrl.dtsi

Thank you for review, I have changed it.

**Review by gmarull — DISMISSED** [review]
pinctrl lgtm

**Review by fkokosinski — DISMISSED** [review]
`boards/riscv`, `dts/riscv` and `soc/riscv` LGTM

**Comment by Cherish-Gww** [comment]
> `boards/riscv`, `dts/riscv` and `soc/riscv` LGTM

@fkokosinski  What should I do next?

**Comment by fkokosinski** [comment]
> > `boards/riscv`, `dts/riscv` and `soc/riscv` LGTM
> 
> @fkokosinski What should I do next?

Hi! Ideally, you'd need to get ACKs from maintainers of the areas that you're modifying here, e.g. gpio, serial etc. You also can ask for help on the `#pr-help` channel on Discord.

**Comment by Cherish-Gww** [comment]
> > > `boards/riscv`, `dts/riscv` and `soc/riscv` LGTM
> > 
> > 
> > @fkokosinski What should I do next?
> 
> Hi! Ideally, you'd need to get ACKs from maintainers of the areas that you're modifying here, e.g. gpio, serial etc. You also can ask for help on the `#pr-help` channel on Discord.

Oh Thanks you very much, I will continue to modify this PR.

**Review by nordicjm — CHANGES_REQUESTED** [review]
Needs to be migrated to hwmv2

**Comment by Cherish-Gww** [comment]
> Needs to be migrated to hwmv2

OK, I will understand hwmv2 and modify my PR as soon as possible.

**Comment by Cherish-Gww** [comment]
> Needs to be migrated to hwmv2

Should I submit my PR to the collab-hwm branch? and change to hwmv2?

**Comment by nordicjm** [comment]
> > Needs to be migrated to hwmv2
> 
> Should I submit my PR to the collab-hwm branch? and change to hwmv2?

Can use what is on hwmv2 as a starting point to see how to transition but we will not be merging new boards/socs into that branch

**Pull request closed without merging**